tezos/context/src/tezedge_context.rs

```diff
@@ -612,23 +629,28 @@ impl IndexApi<TezedgeContext> for TezedgeIndex {
             }
         };
 
-        let mut storage = self.storage.borrow_mut();
-        storage.clear();
+        // TODO: should we always be copying this value? is it possibe
+        // to keep the latest version around and copy only when not different?
+        let index = self.with_clear_storage()?;
 
-        let commit = match self.fetch_commit(hash_id, &mut storage)? {
-            Some(commit) => commit,
-            None => return Ok(None),
-        };
+        let dir_id = {
+            let mut storage = index.storage.borrow_mut();
 
-        let dir_id = match self.fetch_directory(commit.root_hash, &mut storage)? {
-            Some(dir_id) => dir_id,
-            None => return Ok(None),
+            let commit = match self.fetch_commit(hash_id, &mut storage)? {
+                Some(commit) => commit,
+                None => return Ok(None),
+            };
+
+            match self.fetch_directory(commit.root_hash, &mut storage)? {
+                Some(dir_id) => dir_id,
+                None => return Ok(None),
+            }
         };
 
         let tree = WorkingTree::new_with_directory(self.clone(), dir_id);
```

Comment:
```suggestion
        let tree = WorkingTree::new_with_directory(index.clone(), dir_id);
```
This needs to use the newly created index

Reply:
Oh, good catch